packages/loot-core/src/client/accounts/accountsSlice.ts

```diff
@@ -0,0 +1,320 @@
+};
+
+type SetAccountsSyncingPayload = {
+  ids: Array<AccountEntity['id']>;
+};
+
+type MarkAccountFailedPayload = {
+  id: AccountEntity['id'];
+  errorType: string;
+  errorCode: string;
+};
+
+type MarkAccountSuccessPayload = {
+  id: AccountEntity['id'];
+};
+
+type AccountState = {
+  failedAccounts: {
+    [key: AccountEntity['id']]: { type: string; code: string };
+  };
+  accountsSyncing: Array<AccountEntity['id']>;
+};
+
+const accountsSlice = createSlice({
+  name: sliceName,
+  initialState,
+  reducers: {
+    setAccountsSyncing(
+      state,
+      action: PayloadAction<SetAccountsSyncingPayload>,
+    ) {
+      state.accountsSyncing = action.payload.ids;
+    },
+    markAccountFailed(state, action: PayloadAction<MarkAccountFailedPayload>) {
+      state.failedAccounts[action.payload.id] = {
+        type: action.payload.errorType,
+        code: action.payload.errorCode,
+      };
+    },
+    markAccountSuccess(
+      state,
+      action: PayloadAction<MarkAccountSuccessPayload>,
+    ) {
+      delete state.failedAccounts[action.payload.id];
+    },
+  },
+});
+
+type UnlinkAccountPayload = {
+  id: AccountEntity['id'];
+};
+
+export const unlinkAccount = createAppAsyncThunk(
+  'accounts/unlinkAccount',
+  async ({ id }: UnlinkAccountPayload, { dispatch }) => {
+    const { markAccountSuccess } = accountsSlice.actions;
+
+    await send('account-unlink', { id });
+    dispatch(markAccountSuccess({ id }));
+    await dispatch(getAccounts());
+  },
+);
+
+type LinkAccountPayload = {
+  requisitionId: string;
+  account: unknown;
+  upgradingId?: string;
+  offBudget?: boolean;
+};
+
+export const linkAccount = createAppAsyncThunk(
+  `${sliceName}/linkAccount`,
+  async (
+    { requisitionId, account, upgradingId, offBudget }: LinkAccountPayload,
+    { dispatch },
+  ) => {
+    await send('gocardless-accounts-link', {
+      requisitionId,
+      account,
+      upgradingId,
+      offBudget,
+    });
+    await dispatch(getPayees());
+    await dispatch(getAccounts());
+  },
+);
+
+type LinkAccountSimpleFinPayload = {
+  externalAccount: unknown;
+  upgradingId?: AccountEntity['id'];
+  offBudget?: boolean;
+};
+
+export const linkAccountSimpleFin = createAppAsyncThunk(
+  `${sliceName}/linkAccountSimpleFin`,
+  async (
+    { externalAccount, upgradingId, offBudget }: LinkAccountSimpleFinPayload,
+    { dispatch },
+  ) => {
+    await send('simplefin-accounts-link', {
+      externalAccount,
+      upgradingId,
+      offBudget,
+    });
+    await dispatch(getPayees());
+    await dispatch(getAccounts());
+  },
+);
+
+function handleSyncResponse(
+  accountId: AccountEntity['id'],
+  res: {
+    errors: Array<{
+      type: string;
+      category: string;
+      code: string;
+      message: string;
+      internal?: string;
+    }>;
+    newTransactions: Array<TransactionEntity['id']>;
+    matchedTransactions: Array<TransactionEntity['id']>;
+    updatedAccounts: Array<AccountEntity['id']>;
+  },
```

Comment:
This huge type is really unfortunate 😅 is it not defined anywhere else? If it isn't, could be nice to refactor to a type alias at least

Reply:
Good call :)